api/v1beta1/network_disruption.go

```diff
@@ -738,5 +741,11 @@ func TransformCloudSpecToHostsSpec(cloudManager cloudservice.CloudServicesProvid
 		}
 	}
 
-	return hosts, nil
+	if s.Hosts == nil {
+		s.Hosts = hosts
+	} else {
+		s.Hosts = append(s.Hosts, hosts...)
```

Comment:
hmm, that seems okay for now, but it's still dangerous for someone to call this a second time if they don't realize. I see the problem is that the user may _also_ have s.Hosts set, and we can't know if that s.Hosts already contains the cloud hosts.....

Maybe we should move this into the controller instead of the injector? I think this would be fine to include in the disruption_controller

Reply:
we could also make sure we don t add any duplicates in the list of s.Hosts

@aymericDD what do you think?

Reply:
IMO It would be nice if the host can be generated during the call of the `GenerateArgs` function, but it is not possible at the moment. 😢  I am not really fan of having a `boolean` variable passed through a multitude of methods, it works, but not easy to understand the purpose of the variable at the first level. I think, it's okay to populate the list of host in the `disruption_controller.go` line **487** _(just before the iteration of target injections)_:
``` golang
....
	if len(instance.Status.TargetInjections) > 0 && (len(instance.Status.TargetInjections) != len(chaosPodsMap)) {
		r.log.Infow("starting targets injection", "targets", instance.Status.TargetInjections)
	}

	subspec := instance.Spec.DisruptionKindPicker(chaostypes.DisruptionKindNetworkDisruption)
	if reflect.ValueOf(subspec).IsValid() {
		// get the ip ranges of cloud provider services
		if err = instance.Spec.Network.UpdateHostsOnCloudDisruption(r.CloudServicesProvidersManager); err != nil {
			return err
		}
	}

	// iterate through target + existing disruption kind -- to ensure all chaos pods exist
	for targetName, injection := range instance.Status.TargetInjections {
....
```